docs/notes/05-linters-and-tests.md

```diff
@@ -0,0 +1,179 @@
+# Linters and Tests
+
+## Linters
+
+A linter is a tool that analyzes source code and provides feedback on potential errors, coding style violations, and programming best practices. It helps maintain code quality and consistency by enforcing coding standards and identifying potential bugs or issues early in the development process.
+
+### Benefits of Linters
+
+- **Improved Code Quality**: Linters help catch common programming mistakes, maintain consistent coding style, and enforce best practices, leading to cleaner and more reliable code.
+
+- **Early Bug Detection**: By analyzing code statically, linters can identify potential issues before the code is executed, enabling early bug detection and reducing the likelihood of bugs reaching production.
+
+- **Readability and Maintainability**: Linters promote consistent coding style and naming conventions, making the code more readable and maintainable. This improves collaboration among team members and makes it easier to understand and modify the codebase.
+
+- **Codebase Standardization**: Linters enforce coding standards and guidelines across the entire codebase, ensuring a unified coding style and reducing the risk of introducing code inconsistencies or errors.
+
+### Linters Requirements
+
+We require the following linters to be run on the code and included as checks in continuous integration:
+
+- [black](https://black.readthedocs.io/en/stable/)
+- [flake8](https://flake8.pycqa.org/en/latest/): Run command `flake8 src/ccn-template/ --max-complexity 10`
+- [isort](https://pycqa.github.io/isort/)
```

Comment:
I think we should explain what each of these do. And also note that they have configuration options, all of which we will hard-code into `tox` (discussed later)

Reply:
- added more info in the gen_ref_pages.py
- the tox option are explained in the note 05-linters-and-tests.md